src/index.ts

```diff
@@ -0,0 +1,88 @@
+import jwt_decode from "jwt-decode";
+import { Claims, defaultOptions, Options } from "./types";
+import { Configuration, DefaultApi, Target } from "./openapi";
+import { VERSION } from './utils';
+
+export class CfClient {
+
+  private api: DefaultApi;
+  private sdkKey: string;
+  private authToken: string;
+  private environment: string;
+  private configuration: Configuration;
+  private options: Options;
+  private cluster = '1';
+
+  constructor(sdkKey: string, options?: Options) {
+    this.sdkKey = sdkKey
+    this.options = {...defaultOptions, ...options}
+
+    this.configuration = new Configuration({
+      basePath: this.options.baseUrl
+    })
+
+    this.api = new DefaultApi(this.configuration)
+    this.run()
+  }
+
+  private async authenticate(): Promise<string> {
+    const response = await this.api.authenticate({
+      apiKey: this.sdkKey
+    })
+    this.authToken = response.data.authToken
+    this.configuration.accessToken = this.authToken;
+
+    const decoded: Claims = jwt_decode(this.authToken);
+
+    this.environment = decoded.environment;
+    this.cluster = decoded.clusterIdentifier || '1';
+    this.configuration.baseOptions = {
+      'headers': {
+        'User-Agent': `NodeJsSDK/${VERSION}`
+      }
+    }
+    return this.authToken;
+  }
+
+  async fetchData() {
+    const flags = this.api.getFeatureConfig(this.environment, {
+      params: {
+        'cluster': this.cluster
+      }
+    }).then(response => {
+      // prepare cache for storing flags
+      console.log(response.data)
+    }).catch((error: Error) => {
+      console.error('Error loading flags', error)
```

Comment:
Instead of just logging this to the console, it might be worth throwing it so it can be caught in the user's code.

Reply:
Done